rustls/src/server/tls13.rs

```diff
@@ -1499,11 +1500,38 @@ impl State<ServerConnectionData> for ExpectTraffic {
             .request_key_update_and_update_encrypter(common)
     }
 
+    fn into_external_state(self: Box<Self>) -> Result<Box<dyn ExternalState + 'static>, Error> {
+        Ok(self)
+    }
+
     fn into_owned(self: Box<Self>) -> hs::NextState<'static> {
         self
     }
 }
 
+impl ExternalState for ExpectTraffic {
+    fn update_secrets(&mut self, dir: Direction) -> Result<ConnectionTrafficSecrets, Error> {
+        let side = match dir {
+            Direction::Transmit => Side::Server,
+            Direction::Receive => Side::Client,
+        };
```

Comment:
Nit: suggest inlining these matches everywhere.

Reply:
I'm not sure where you want to inline these? This match statement is different based on whether we are on the server or client side of the connection and we don't have that info in `KernelConnection`

Reply:
I meant like:

```rust
self.key_schedule.refresh_traffic_secret(match dir {
    ..
})
```